src/erlcloud_aws.erl

```diff
@@ -194,15 +202,23 @@ format_timestamp({{Yr, Mo, Da}, {H, M, S}}) ->
 default_config() ->
     case get(aws_config) of
         undefined ->
-            #aws_config{access_key_id=os:getenv("AWS_ACCESS_KEY_ID"),
-                        secret_access_key=os:getenv("AWS_SECRET_ACCESS_KEY")};
+            AccessKeyId = case os:getenv("AWS_ACCESS_KEY_ID") of
+                              false -> "";
```

Comment:
Is there a reason that access_key_id needs to be a string? It seems like `undefined` would be a more conventional value if it isn't set, and would be less likely to break things like guard tests that look for lists.